src/Common/src/System/Security/Cryptography/AsnReader.cs

```diff
@@ -432,6 +432,37 @@ public ReadOnlyMemory<byte> GetIntegerBytes(Asn1Tag expectedTag)
             return contents;
         }
 
+        public BigInteger GetInteger() => GetInteger(Asn1Tag.Integer);
+
+        public BigInteger GetInteger(Asn1Tag expectedTag)
+        {
+            ReadOnlyMemory<byte> contents =
+                GetIntegerContents(expectedTag, UniversalTagNumber.Integer, out int headerLength);
+
+            // TODO: Split this for netcoreapp/netstandard to use the Big-Endian BigInteger parsing
+            byte[] tmp = ArrayPool<byte>.Shared.Rent(contents.Length);
+            BigInteger value;
+
+            try
+            {
+                byte fill = (contents.Span[0] & 0x80) == 0 ? (byte)0 : (byte)0xFF;
+                // Fill the unused portions of tmp with positive or negative padding.
+                new Span<byte>(tmp, contents.Length, tmp.Length - contents.Length).Fill(fill);
+                contents.CopyTo(tmp);
+                // Convert to Little-Endian.
+                AsnWriter.Reverse(new Span<byte>(tmp, 0, contents.Length));
+                value = new BigInteger(tmp);
+            }
+            finally
+            {
+                Array.Clear(tmp, 0, contents.Length);
```

Comment:
Leaks the high bit of _contents_ since the filler 0x00 / 0xFF bytes are never cleared.